python-checks/src/main/java/org/sonar/python/checks/tests/ImplicitlySkippedTestCheck.java

```diff
@@ -121,20 +124,20 @@ private static ReturnStatement getReturnStatementFromFirstIfStatement(FunctionDe
   }
 
   private static boolean containsAssertion(FunctionDef functionDef) {
-    List<String> parentClasses = getParentClassFromFunctionDef(functionDef);
-    AssertionVisitor assertVisitor = new AssertionVisitor(parentClasses);
+    Optional<TestFramework> testFramework = getParentClassTestFrameworkFromFunctionDef(functionDef);
+    AssertionVisitor assertVisitor = new AssertionVisitor(testFramework.orElse(emptyFramework).supportedAssertMethods);
     functionDef.accept(assertVisitor);
     return assertVisitor.hasAnAssert;
   }
 
-  private static List<String> getParentClassFromFunctionDef(FunctionDef functionDef) {
+  private static Optional<TestFramework> getParentClassTestFrameworkFromFunctionDef(FunctionDef functionDef) {
     List<String> libraries = new ArrayList<>();
     ClassDef classDef = (ClassDef) TreeUtils.firstAncestorOfKind(functionDef, Tree.Kind.CLASSDEF);
     if (classDef != null) {
       libraries.addAll(getInheritedClassesFQN(classDef));
     }
 
-    return libraries;
+    return testFrameworks.stream().filter(testFramework -> testFramework.matchAnyProvidedClasses(libraries)).findFirst();
```

Comment:
I'd suggest replacing this with:

```
    return testFrameworks.stream()
      .filter(testFramework -> testFramework.matchAnyProvidedClasses(libraries))
      .findFirst()
      .map(t -> t.supportedAssertMethods)
      .orElseGet(Collections::emptySet);
```

and make the method return a set of assertion methods instead of an optional of test framework.
That way, you can get rid of the `emptyFramework` you had to define.